pallets/extended-assets/src/lib.rs

```diff
@@ -341,6 +341,35 @@ pub mod pallet {
 
             Ok(())
         }
+
+        /// Marks an asset as regulated, representing that the asset will only operate between KYC-verified wallets.
+        ///
+        /// ## Parameters
+        ///
+        /// - `origin`: The origin of the transaction.
+        /// - `asset_id`: The identifier of the asset.
+        #[pallet::call_index(4)]
+        #[pallet::weight(<T as Config>::WeightInfo::regulate_asset())]
+        pub fn regulate_asset(origin: OriginFor<T>, asset_id: AssetIdOf<T>) -> DispatchResult {
+            let who = ensure_signed(origin)?;
```

Comment:
Do we want to allow this for `sudo` as well?

Reply:
There is no such requirement so, I don't think so. it's just in the scope of asset owner.

Reply:
We have to address the question to requirements.

Reply:
@AndreyKortosov 
Can you please address this in requirements?

The question:
Do we need the sudo (root) account to have access to regulate any asset in addition to AssetOwner account?


cc. @wer1st

Reply:
@HaidarJbeily7 AssetOwner account will regulate any asset. Am I right?
What's the purpose of sharing access to another account?

Reply:
@AndreyKortosov 
AssetOwner account will be able to regulate **the assets that the account owns**.
There is a sudo account which has superior accesses. So, we need to address in requirements if we need to share access with this sudo account to this extrinsic, or simply not?

@vovac12 Do you think there is a use case that we would need sudo access?

Reply:
Ok